libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/headline/PluggableHeadline.tsx

```diff
@@ -299,21 +301,47 @@ export class PluggableHeadline extends AbstractPluggableVisualization {
         }, []);
     }
 
-    private initializeProperties(visualizationProperties: IVisualizationProperties): void {
-        const controls = visualizationProperties?.controls;
-
-        const supportedProperties = getSupportedPropertiesControls(controls, this.supportedPropertiesList);
-        const initialProperties = {
-            supportedProperties: {
+    protected updateInstanceProperties(
```

Comment:
Can this method be broken down into smaller methods and be better explained what it does and why? It is not immediately obvious what this code does.

Reply:
Done. I hope it's clearer now.